custom_components/apple_tv/__init__.py

```diff
@@ -339,3 +365,31 @@ def address_updated(self, address):
             data={**self.config_entry.data, CONF_ADDRESS: address},
         )
         self.hass.add_job(update_entry, self.config_entry)
+
+    @staticmethod
+    async def async_options_updated(hass, entry):
+        """Triggered by config entry options updates."""
+        hass.data[DOMAIN][entry.unique_id]._is_pwr_mgmt_on = entry.options[CONF_PWR_MGMT]
+        if entry.options[CONF_PWR_MGMT] and hass.data[DOMAIN][entry.unique_id].atv:
```

Comment:
I'm not sure about this functionality (from here and until end of function), is there any particular reason for doing this?

Reply:
@postlund it's covering the following situations...
For example:
Apple TV power was On and HASS Remote/Media Control switch was Off (`atv = None`). If you changing Option to `Allow to Home Assistant to use power management functionality` this code will create `atv` instance and will update the `power_state` to the real one.

The same goes for the vice-versa scenario.

I hope it's clear.

Reply:
@postlund all good or you have more questions? 
can I progress with the tests? :)

Reply:
I'm still not a 100% sure about having the callback trigger power on/off automatically like but. But I'll leave it for the Home Assistant devs that does the review later on. We can say that it's ok! Carry on with the tests!